cmd/generatetf/generator/generator.go

```diff
@@ -0,0 +1,273 @@
+	for r, e := range entries {
+		if e.APIType == apiType {
+			return r, &e
+		}
+	}
+
+	return "", nil
+}
+
+// generateConfigBlock generate a source or destination config block from an API config JSON
+// and a corresponding ConfigMeta.
+func generateConfigBlock(config json.RawMessage, cm *configs.ConfigMeta) (*hclwrite.Block, error) {
+	// get state representation of config as map[string]interface{}
+	state, err := cm.APIToState(string(config))
+	if err != nil {
+		return nil, fmt.Errorf("could not convert API config to terraform state: %w", err)
+	}
+
+	var stateMap map[string]interface{}
+	if err := json.Unmarshal([]byte(state), &stateMap); err != nil {
+		return nil, err
+	}
+
+	block, err := generateBlock("config", stateMap, cm.ConfigSchema)
+	if err != nil {
+		return nil, fmt.Errorf("could not generate config block: %w", err)
+	}
+
+	return block, nil
+}
+
+// generateBlock generate blocks with specified name and data which conform to the provided config schema.
+// data should be unmarshaled from a state JSON object.
+func generateBlock(name string, data map[string]interface{}, configSchema map[string]*schema.Schema) (*hclwrite.Block, error) {
+	block := hclwrite.NewBlock(name, []string{})
+	body := block.Body()
+
+	// go does not garantee the order of range in maps, we sort the keys so that the output is predictable
+	var keys []string
+	for k := range data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		v := data[k]
+		if sch, ok := configSchema[k]; ok {
+			// state objects are actually lists with a single object item
+			if r, ok := sch.Elem.(*schema.Resource); ok && sch.Type == schema.TypeList && sch.ConfigMode != schema.SchemaConfigModeAttr {
+				if l, ok := v.([]interface{}); ok && len(l) > 0 {
+					if value, ok := l[0].(map[string]interface{}); ok {
+						kBlock, err := generateBlock(k, value, r.Schema)
+						if err != nil {
+							return nil, err
+						}
+						body.AppendBlock(kBlock)
+					}
+				}
+			} else {
+				body.SetAttributeValue(k, ctyValue(v))
+			}
+		}
+	}
+
+	return block, nil
+}
+
+// ctyValue converts any value to a cty.Value which can be passed as values to HCL attributes
+func ctyValue(x interface{}) cty.Value {
+	switch v := x.(type) {
+	case string:
+		return cty.StringVal(v)
+
+	case bool:
+		return cty.BoolVal(v)
+
+	case int:
+	case int16:
+	case int32:
+	case int64:
+		return cty.NumberIntVal(v)
+
+	case float32:
+	case float64:
+		return cty.NumberFloatVal(v)
+
+	case []interface{}:
+		var values []cty.Value
+		for _, i := range v {
+			values = append(values, ctyValue(i))
+		}
+		return cty.ListVal(values)
+
+	case map[string]interface{}:
+		values := map[string]cty.Value{}
+		for k, i := range v {
+			values[k] = ctyValue(i)
+		}
+		return cty.ObjectVal(values)
+	}
+
+	return cty.EmptyObjectVal
+}
+
+func generateConnection(connection client.Connection) (*hclwrite.Block, error) {
```

Comment:
shall we only create for connections btw supported sources and destinations?

Reply:
That is also a good idea